apps/els_dap/test/els_dap_general_provider_SUITE.erl

```diff
@@ -0,0 +1,408 @@
+-module(els_dap_general_provider_SUITE).
+
+%% CT Callbacks
+-export([
+    suite/0,
+    init_per_suite/1,
+    end_per_suite/1,
+    init_per_testcase/2,
+    end_per_testcase/2,
+    groups/0,
+    all/0
+]).
+
+%% Test cases
+-export([
+    initialize/1,
+    launch_mfa/1,
+    configuration_done/1,
+    configuration_done_with_breakpoint/1,
+    frame_variables/1,
+    navigation_and_frames/1,
+    set_variable/1
+]).
+
+%% TODO: cleanup after dropping support for OTP 21 and 22
+-compile({no_auto_import, [atom_to_binary/1, binary_to_atom/1]}).
+
+%%==============================================================================
+%% Includes
+%%==============================================================================
+-include_lib("common_test/include/ct.hrl").
+-include_lib("stdlib/include/assert.hrl").
+
+%%==============================================================================
+%% Types
+%%==============================================================================
+-type config() :: [{atom(), any()}].
+
+%%==============================================================================
+%% CT Callbacks
+%%==============================================================================
+-spec suite() -> [tuple()].
+suite() ->
+    [{timetrap, {seconds, 30}}].
+
+-spec all() -> [atom()].
+all() ->
+    els_dap_test_utils:all(?MODULE).
+
+-spec groups() -> [atom()].
+groups() ->
+    [].
+
+-spec init_per_suite(config()) -> config().
+init_per_suite(Config) ->
+    Config.
+
+-spec end_per_suite(config()) -> ok.
+end_per_suite(_Config) ->
+    ok.
+
+-spec init_per_testcase(atom(), config()) -> config().
+init_per_testcase(TestCase, Config) when
+    TestCase =:= undefined orelse
```

Comment:
We could use an auxiliary function here, instead of calling back the `init_per_testcase/2` with an `undefined` atom.